hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ClientDiscoverySpiTest.java

```diff
@@ -374,6 +375,26 @@ public void testMulticastDiscoveryEnabledNoLocalhost() {
         HazelcastClient.newHazelcastClient(clientConfig);
     }
 
+    @Test
+    public void testCustomDiscovery_startCalledOnce() {
+        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
+
+        ClientConfig config = new ClientConfig();
+        config.setProperty(ClusterProperty.DISCOVERY_SPI_ENABLED.getName(), "true");
+
+        final DiscoveryService discoveryService = mock(DiscoveryService.class);
+        when(discoveryService.discoverNodes()).thenReturn(Collections.singletonList(new SimpleDiscoveryNode(instance.getCluster().getLocalMember().getAddress())));
+        DiscoveryServiceProvider discoveryServiceProvider = arg0 -> discoveryService;
+        ClientNetworkConfig networkConfig = config.getNetworkConfig();
+        config.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(0);
+        networkConfig.getDiscoveryConfig().addDiscoveryStrategyConfig(new DiscoveryStrategyConfig());
+        networkConfig.getDiscoveryConfig().setDiscoveryServiceProvider(discoveryServiceProvider);
+
+        HazelcastClient.newHazelcastClient(config);
```

Comment:
why does this file use real network?

Reply:
I don't think it should. I believe this test was added before we have added the mock network support for client tests.

I didn't think much about it while writing the test, but let me make it use the mock network